backends/postgres.go

```diff
@@ -132,14 +124,34 @@ func NewPostgres(authOpts map[string]string, logLevel log.Level, hasher hashing.
 	//Build the dsn string and try to connect to the db.
 	connStr := fmt.Sprintf("user=%s password=%s dbname=%s host=%s port=%s", postgres.User, postgres.Password, postgres.DBName, postgres.Host, postgres.Port)
 
-	if (postgres.SSLMode == "verify-ca" || postgres.SSLMode == "verify-full") && checkSSL {
-		connStr = fmt.Sprintf("%s sslmode=verify-ca sslcert=%s sslkey=%s sslrootcert=%s", connStr, postgres.SSLCert, postgres.SSLKey, postgres.SSLRootCert)
-	} else if postgres.SSLMode == "require" {
+	if postgres.SSLMode == "verify-ca" {
+		connStr = fmt.Sprintf("%s sslmode=verify-ca", connStr)
+	}
+
+	if postgres.SSLMode == "verify-full" {
+		connStr = fmt.Sprintf("%s sslmode=verify-full", connStr)
+	}
+
+	if postgres.SSLMode == "require" {
 		connStr = fmt.Sprintf("%s sslmode=require", connStr)
-	} else {
+	}
+
+	if postgres.SSLMode == "disable" {
 		connStr = fmt.Sprintf("%s sslmode=disable", connStr)
 	}
 
+	if len(postgres.SSLRootCert) > 0 {
```

Comment:
Zero value of string is and empty one, please compare to that instead of checking length.